homeassistant/components/reolink/entity.py

```diff
@@ -3,47 +3,63 @@
 
 from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
 from homeassistant.helpers.entity import DeviceInfo
-from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 from . import ReolinkData
 from .const import DOMAIN
 
 
-class ReolinkCoordinatorEntity(CoordinatorEntity):
+class ReolinkBaseCoordinatorEntity(CoordinatorEntity):
     """Parent class for Reolink hardware camera entities."""
 
-    def __init__(self, reolink_data: ReolinkData, channel: int) -> None:
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, reolink_data: ReolinkData, coordinator: DataUpdateCoordinator
+    ) -> None:
         """Initialize ReolinkCoordinatorEntity for a hardware camera."""
-        coordinator = reolink_data.device_coordinator
         super().__init__(coordinator)
 
         self._host = reolink_data.host
-        self._channel = channel
 
         http_s = "https" if self._host.api.use_https else "http"
-        conf_url = f"{http_s}://{self._host.api.host}:{self._host.api.port}"
+        self._conf_url = f"{http_s}://{self._host.api.host}:{self._host.api.port}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._host.unique_id)},
+            connections={(CONNECTION_NETWORK_MAC, self._host.api.mac_address)},
+            name=self._host.api.nvr_name,
+            model=self._host.api.model,
+            manufacturer=self._host.api.manufacturer,
+            hw_version=self._host.api.hardware_version,
+            sw_version=self._host.api.sw_version,
+            configuration_url=self._conf_url,
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._host.api.session_active and super().available
+
+
+class ReolinkCoordinatorEntity(ReolinkBaseCoordinatorEntity):
```

Comment:
I'm confused what or why they are different. The class descriptions are exactly the same?

Reply:
Sorry, I forgot to update the docstrings.
I now put a (hopefully) clear discription in the docstrings:
`ReolinkBaseCoordinatorEntity` is for entities that control the NVR
`ReolinkCoordinatorEntity` is for entities that control a camera connected to the NVR

A camera connected directly to HomeAssistant withouth using a NVR is in the reolink API basically a NVR with a single channel that has the camera connected to that channel.

Some exotic models like the the [dual lens cameras](https://store.reolink.com/dual-lens-cameras) when connected directly to HomeAssistant are then basically a NVR with 2 channels with each of the lenses connected to one of the channels.